contracts/ibc-forwarder/src/contract.rs

```diff
@@ -87,9 +87,15 @@ pub fn execute(
     info: MessageInfo,
     msg: ExecuteMsg,
 ) -> NeutronResult<Response<NeutronMsg>> {
-    match msg {
-        ExecuteMsg::DistributeFallback { coins } => try_distribute_fallback(deps, env, info, coins),
-        ExecuteMsg::Tick {} => try_tick(deps, env, info),
+    match (CONTRACT_STATE.load(deps.storage)?, msg) {
+        // ticks coming in to ibc forwarder in Instantiaed state will attempt to register the ICA
+        (ContractState::Instantiated, ExecuteMsg::Tick {}) => try_register_ica(deps, info, env),
+        // ticks coming in to ibc forwarder in IcaCreated state will attempt to forward the funds
+        (ContractState::IcaCreated, ExecuteMsg::Tick {}) => try_forward_funds(env, info, deps),
```

Comment:
The `try_forward_funds`, `try_register_ica` & `try_distribute_fallback` each have different argument orders.